src/search/pdbs/pattern_collection_generator_multiple_random.cc

```diff
@@ -0,0 +1,91 @@
+#include "pattern_collection_generator_multiple_random.h"
+#include "pattern_database.h"
+#include "random_pattern.h"
+#include "utils.h"
+
+#include "../option_parser.h"
+#include "../plugin.h"
+#include "../task_proxy.h"
+
+#include "../task_utils/causal_graph.h"
+
+#include "../utils/countdown_timer.h"
+#include "../utils/logging.h"
+#include "../utils/markup.h"
+#include "../utils/rng.h"
+#include "../utils/rng_options.h"
+
+#include <vector>
+
+using namespace std;
+
+namespace pdbs {
+PatternCollectionGeneratorMultipleRandom::PatternCollectionGeneratorMultipleRandom(
+    options::Options &opts)
+    : PatternCollectionGeneratorMultiple(opts),
+      bidirectional(opts.get<bool>("bidirectional")) {
+}
+
+string PatternCollectionGeneratorMultipleRandom::get_name() const {
+    return "Random Patterns";
+}
+
+void PatternCollectionGeneratorMultipleRandom::initialize(
+    const shared_ptr<AbstractTask> &task) {
+    // Compute CG neighbors once. They will be shuffled after each use.
+    cg_neighbors = compute_cg_neighbors(task, bidirectional);
+}
+
+PatternInformation PatternCollectionGeneratorMultipleRandom::compute_pattern(
+    int max_pdb_size,
+    double max_time,
+    const shared_ptr<utils::RandomNumberGenerator> &rng,
+    const shared_ptr<AbstractTask> &task,
+    FactPair goal,
+    unordered_set<int> &&) {
+    // TODO: add support for blacklisting in single RCG?
```

Comment:
While this is not yet supported, should we check for it here and warn the user if it is used?